lib/pure/parsecsv.nim

```diff
@@ -75,8 +75,13 @@ proc raiseEInvalidCsv(filename: string, line, col: int,
   e.msg = filename & "(" & $line & ", " & $col & ") Error: " & msg
   raise e
 
+template `||`[T](a, default: T): T =
+  ## `||` is a helper to allow replacing nilable types with a default
+  if a.isNil(): default else: a
+
 proc error(my: CsvParser, pos: int, msg: string) =
-  raiseEInvalidCsv(my.filename, my.lineNumber, getColNumber(my, pos), msg)
+  raiseEInvalidCsv(my.filename || "", my.lineNumber, getColNumber(my, pos), msg)
```

Comment:
I made the nil check in the `raiseEInvalidCsv` proc instead so that a nil `filename` or `msg` will never cause problems if raising an exception from elsewhere.